ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/manager/EthSchedulerTest.java

```diff
@@ -204,4 +207,19 @@ public void timeout_cancelsTaskWhenResultIsCancelled() {
     assertThat(task.isSucceeded()).isFalse();
     assertThat(task.isCancelled()).isTrue();
   }
+
+  @Test
+  public void scheduleTxWorkerTask_completesWhenScheduledTaskCompletes() {
+    final MockRunnable runnable = Mockito.mock(MockRunnable.class);
+    final MockExecutorService mockTxWorker = ethScheduler.txWorkerExecutor();
+    ethScheduler.scheduleTxWorkerTask(runnable);
+    MatcherAssert.assertThat(mockTxWorker.getFutures(), hasSize(1));
+    verify(runnable).run();
+  }
+
+  static class MockRunnable implements Runnable {
```

Comment:
This class is redundant - just mock `Runnable.class` directly.

Reply:
Removed.